Assets/Scripts/Audio/AudioWaveformDrawer.cs

```diff
@@ -0,0 +1,317 @@
+        /// </summary>
+        /// <param name="position">Position to calculate the song time.</param>
+        /// <returns>Song time (float)</returns>
+        public float CalculateSongTimeBasedOnPosition(Vector2 position)
+        {
+            float xPos = position.x - _waveformPositionOffset.x;
+            xPos = Mathf.Clamp(xPos, 0, _songWidth);
+
+            return xPos * _audioClipDuration / _songWidth; ;
+        }
+
+        /// <summary>
+        /// Log all available keybinds to the console.
+        /// </summary>
+        public void LogKeybinds()
+        {
+            Debug.Log("<b>Available Keybinds:</b>");
+            Debug.Log("Pause song key: " + _pauseKey);
+            Debug.Log(" "); //empty line
+
+            Debug.Log("Scrub forward key: " + _forwardKey);
+            Debug.Log("Scrub backward Key: " + _backwardKey);
+            Debug.Log(" "); //empty line
+
+            Debug.Log("Speed up Key: " + _speedUpKey);
+            Debug.Log("Speed down Key: " + _speedDownKey);
+            Debug.Log(" "); //empty line
+
+            Debug.Log("Go to start Key: " + _homeKey);
+            Debug.Log("Go to end Key: " + _endKey);
+
+            Debug.Log(" "); //empty line
+            Debug.Log(" "); //empty line
+            onShowKeyBinds?.Invoke();
+        }
+
+        /// <summary>
+        /// Method for initialzing the waveform drawer for a specific AudioClip.
+        /// </summary>
+        /// <param name="audioClip">Audio clip to draw.</param>
+        public void InitializeDrawer(AudioClip audioClip)
+        {
+            Destroy(_waveformTexture);
+
+            AudioClip lastClip = audioSource.clip;
+            audioSource.clip = audioClip;
+
+            onSongChanged?.Invoke(audioClip.name, lastClip != null ? lastClip.name : "Null");
+
+            _dataSamples = new float[audioSource.clip.samples * audioSource.clip.channels]; // create an array to store the audio data
+            audioSource.clip.GetData(_dataSamples, 0); //get audio data from the audio clip
+
+            _waveformTexture = new Texture2D(_textureWidth, _textureHeight, TextureFormat.RGBA32, false); //prepare the waveform texture
+            _textureColors = new Color[_waveformTexture.width * _waveformTexture.height]; //create an array to store the texture pixel data
+
+            GenerateWaveformTexture();
+            Renderer renderer = GetComponent<Renderer>();
+
+            //set the offset for the texture
+            _waveformPositionOffset = new Vector2(-(_textureWidth + _textureWidth / _renderDownScaleModifier), _curserYPosition);
+            _songWidth = Mathf.Abs(_waveformPositionOffset.x * 2);
+
+            if (renderer != null)
+                renderer.material.mainTexture = _waveformTexture;
+
+            _audioClipDuration = audioSource.clip.length;
+
+            if (isPaused) 
+                audioSource.Pause();
+            else 
+                audioSource.Play();
```

Comment:
```suggestion
            if (isPaused && audioSource.IsPlaying()) 
                audioSource.Pause();
            else if(!audioSource.IsPlaying())
                audioSource.Play();
```

Reply:
Why? This is not really necessary since I only want to know if the user has already paused the track.